engine/schema/src/main/java/org/apache/cloudstack/backup/dao/BackupDaoImpl.java

```diff
@@ -141,32 +143,37 @@ public List<Backup> syncBackups(Long zoneId, Long vmId, List<Backup> externalBac
-
-        BackupResponse response = new BackupResponse();
-        response.setId(backup.getUuid());
-        response.setVmId(vm.getUuid());
-        response.setVmName(vm.getHostName());
-        response.setExternalId(backup.getExternalId());
-        response.setType(backup.getType());
-        response.setDate(backup.getDate());
-        response.setSize(backup.getSize());
-        response.setProtectedSize(backup.getProtectedSize());
-        response.setStatus(backup.getStatus());
-        response.setVolumes(new Gson().toJson(vm.getBackupVolumeList().toArray(), Backup.VolumeInfo[].class));
-        response.setBackupOfferingId(offering.getUuid());
-        response.setBackupOffering(offering.getName());
-        response.setAccountId(account.getUuid());
-        response.setAccount(account.getAccountName());
-        response.setDomainId(domain.getUuid());
-        response.setDomain(domain.getName());
-        response.setZoneId(zone.getUuid());
-        response.setZone(zone.getName());
-        response.setObjectName("backup");
-        return response;
+        try {
+            VMInstanceVO vm = vmInstanceDao.findByIdIncludingRemoved(backup.getVmId());
+            AccountVO account = accountDao.findByIdIncludingRemoved(vm.getAccountId());
+            DomainVO domain = domainDao.findByIdIncludingRemoved(vm.getDomainId());
+            DataCenterVO zone = dataCenterDao.findByIdIncludingRemoved(vm.getDataCenterId());
+            BackupOffering offering = backupOfferingDao.findByIdIncludingRemoved(backup.getBackupOfferingId());
+
+            BackupResponse response = new BackupResponse();
+            response.setId(backup.getUuid());
+            response.setVmId(vm.getUuid());
+            response.setVmName(vm.getHostName());
+            response.setExternalId(backup.getExternalId());
+            response.setType(backup.getType());
+            response.setDate(backup.getDate());
+            response.setSize(backup.getSize());
+            response.setProtectedSize(backup.getProtectedSize());
+            response.setStatus(backup.getStatus());
+            response.setVolumes(GSON.toJson(vm.getBackupVolumeList().toArray(), Backup.VolumeInfo[].class));
+            response.setBackupOfferingId(offering.getUuid());
+            response.setBackupOffering(offering.getName());
+            response.setAccountId(account.getUuid());
+            response.setAccount(account.getAccountName());
+            response.setDomainId(domain.getUuid());
+            response.setDomain(domain.getName());
+            response.setZoneId(zone.getUuid());
+            response.setZone(zone.getName());
+            response.setObjectName("backup");
+            return response;
+        } catch (Exception e) {
+            LOGGER.error(String.format("Failed to create backup response from Backup [id: %s, vmId: %s] due to: [%s].", backup.getId(), backup.getVmId(), e.getMessage()), e);
+            return null;
+        }
```

Comment:
Does this whole block need to be guarded with a Pokemon catch?  
Which methods are we expecting to throw exceptions?  
We can probably change the catch to a couple of exceptions.